woocommerce/woocommerce-ios
# Adopt new way of getting notification data in PushNotificationBackgroundSynchronizer

Closes: #13852


## Description


Since we might already have `note` in the push notification (parsed from `note_full_data` part of the payload) we can check if we already have `orderID` in it without the need of syncronizing all the notifications and avoid extra network calls.
If we have the `orderID` we can go directly to syncronizing that **exact** order.

In this PR we use that ability and we use `orderID` from the `note` if we have that property set/loaded.

## Testing information


Since this PR changes are not visible from the UI, testing can be done by adding some breakpoints in the `PushNotificationBackgroundSynchronizer`:
- add breakpoint to line 40 -> `orderID = Int64(noteOrderID)`
- run the app from Xcode
- put the app in the background
- get a new order notification
- check that `sync()` function in `PushNotificationBackgroundSynchronizer` is called and debugger stops on that breakpoint at line 40
- check that the notification has the proper `orderID

<img width="1624" alt="Screenshot 2024-09-12 at 11 39 12" src="https://github.com/user-attachments/assets/b800bb6e-96de-4999-9b2d-b65e69f9d09c">

Main breakpoint:

<img width="1624" alt="Screenshot 2024-09-12 at 11 39 17" src="https://github.com/user-attachments/assets/841a2e8b-8a7f-4b83-bfc9-0e847035eef5">

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
